rtos/source/ThisThread.cpp

```diff
@@ -251,4 +251,32 @@ const char *get_name()
 #endif
 }
 
+osPriority_t ThisThread::get_priority()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osPriorityError;
+    }
+
+    return static_cast<osPriority_t>(osThreadGetPriority(id));
+#else
+    return static_cast<osPriority_t>(osPriorityError);
```

Comment:
We might also want to say osPriorityNormal for baremetal Mbed OS, rather than an error.

Reply:
`osPriorityNormal` refers to 24, are we to use something else?

Reply:
osPriorityNormal is the default priority. It makes sense as a sane default to return in baremetal, at least to me.